### PR TITLE
Migrate GetAttesterSlashings to MigratingEndpointAdapter

### LINKED ISSUE #5460: Migrate GetAttesterSlashings to MigratingEndpointAdapter

### PR DESCRIPTION
**Pull request by courtneyeh**
fixes #5460 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetAttesterSlashings.java [hunk 1]
```diff
@@ -13,36 +13,55 @@
...
+        EndpointMetadata.get(ROUTE)
+            .operationId("getAttesterSlashings")
+            .summary("Get Attester Slashings")
+            .description(
+                "Retrieves attester slashings known by the node but not necessarily incorporated into any block.")
+            .tags(TAG_BEACON)
+            .response(
+                SC_OK,
+                "Request successful",
+                getResponseType(spec.getGenesisSpecConfig())) // TODO check correct
```
Comment (courtneyeh) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetAttesterSlashings.java]: ^^ Just need to check this is the correct approach to getting the `SpecConfig`

Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetAttesterSlashings.java]: I'd probably pass just the spec and then if we need to get different configs at a future date it'll just be an implementation detail, but this is fine too...

**Review by rolfyone — APPROVED** [review]
LGTM, can change specConfig or we can change it later if we ever need to, I'm not super concerned about that.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetAttesterSlashings.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetAttesterSlashings.java]: either way the TODO needs to disappear :)

**Pull request merged**